lib/class-wp-rest-pattern-directory-controller.php

```diff
@@ -85,7 +85,18 @@ public function get_items_permissions_check( $request ) { // phpcs:ignore Variab
 	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
 	 */
 	public function get_items( $request ) {
-		$query_args  = array();
+		/*
+		 * Include an unmodified `$wp_version`, so the API can craft a response that's tailored to
+		 * it. Some plugins modify the version in a misguided attempt to improve security by
+		 * obscuring the version, which can cause invalid requests.
+		 */
+		require ABSPATH . WPINC . '/version.php';
+
+		$query_args = array(
+			'locale'     => get_user_locale(),
+			'wp-version' => $wp_version, // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable -- it's defined in `version.php` above.
```

Comment:
🤔 , i wonder if this should be the _Gutenberg_ version, rather than the _WP_ version? Or maybe we'll want both?

Probably safest to just add the G version too, since we haven't worked out exactly what info the API will need to respond to different compatibility concerns

cc @ryelle , @dd32 , @coreymckrill

Reply:
Is there a way to get the "editor version" in a core-friendly way? This endpoint is for compatibility only, and won't load in 5.8+ because the core class `WP_REST_Pattern_Directory_Controller` exists (unless something changes). So if we add that here, we'll want to add the same arg to core.

I do think the editor version would be a more relevant value for figuring out pattern compatibility, but I don't see the harm in sending both, if editor is possible.

Reply:
That's a good idea. I looked through Core's code and [the docs for the release process](https://make.wordpress.org/core/handbook/about/release-cycle/block-editor-release-process-for-major-releases/), though, and didn't find anything. It's really late in the cycle to be making any changes to Core anyway.

Reply:
I went ahead and added a separate field for the Gutenberg version (and rebased) in b70a3526ead6364831270a01735604b5d266d892